app/assets/javascripts/components/common/course_navbar.jsx

```diff
@@ -84,6 +84,12 @@ const CourseNavbar = ({ course, location, currentUser, courseLink }) => {
     );
   }
 
+  // ///////////////////////
+  // Prefix for home wiki /
+  // //////////////////////
+  let string_article_prefix = course.home_wiki.project === 'wikidata' ? 'wikidata' : 'others';
+  string_article_prefix = `${course.string_prefix}.${string_article_prefix}`;
```

Comment:
Interesting. With this approach, we end up with two different branches for any given label: course type (course string prefix), and wiki (article prefix).

I think the terminology we use in place of "Article" will be independent of course type, so branching on course string prefix results in duplication.

I think a better approach would be make sure that all the article labels are not in the courses/courses_generic message trees, and instead move any you find from there to `articles` message tree, and then have alternative branches for `pages` and `items`.

The `article_string_prefix` is probably a property that should come from the `Wiki` model and be passed from the server to the frontend.

Reply:
I have created a wiki_prefix in the backend instead of initializing itself in course_navbar component as I was doing last time. And using that data I'm fetching labels under Articles which will be independent of course type.
@ragesoss Please review.